### PR TITLE
Fix warnings in unit-tests tests

### LINKED ISSUE #6155: Fix warnings in the tests
Currently, there are following warnings when running tests:
```
test/test_models.py::test_quantized_classification_model[googlenet]
  /root/project/torchvision/models/googlenet.py:47: FutureWarning: The default weight initialization of GoogleNet will be changed in future releases of torchvision. If you wish to keep the old behavior (which leads to long initialization times due to scipy/scipy#11299), please set init_weights=True.
    warnings.warn(

test/test_models.py::test_quantized_classification_model[inception_v3]
  /root/project/torchvision/models/inception.py:43: FutureWarning: The default weight initialization of inception_v3 will be changed in future releases of torchvision. If you wish to keep the old behavior (which leads to long initialization times due to scipy/scipy#11299), please set init_weights=True.
    warnings.warn(

test/test_datasets.py::VOCDetectionTestCase::test_transforms
  /root/project/torchvision/datasets/voc.py:82: UserWarning: Accessing the test image set of the year 2007 with year='2007-test' is deprecated since 0.12 and will be removed in 0.14. Please use the combination year='2007' and image_set='test' instead.
    warnings.warn(
```

Source: https://app.circleci.com/pipelines/github/pytorch/vision/18272/workflows/46f9c352-ff53-4d87-a894-b124e49d0773/jobs/1478749

Let's fix these tests such that warnings are not shown anymore.

cc @pmeier

### PR DESCRIPTION
**Pull request by puhuk**
Fixes #6155

### CODE REVIEW

**Comment by vfdev-5** [comment]
@puhuk ci failures are related

File: test/test_models.py [hunk 1]
```diff
@@ -870,6 +870,7 @@ def test_quantized_classification_model(model_fn):
         "num_classes": 5,
         "input_shape": (1, 3, 224, 224),
         "quantize": True,
+        "init_weights": True,
```
Comment (vfdev-5) [inline test/test_models.py:873]: please remove this line to fix the CI

File: torchvision/models/quantization/mobilenetv2.py [hunk 2]
```diff
@@ -46,6 +46,8 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         Args:
            Inherits args from floating point MobileNetV2
         """
+        if "init_weights" not in kwargs:
+            kwargs["init_weights"] = True
```
Comment (vfdev-5) [inline torchvision/models/quantization/mobilenetv2.py:50]: MobileNetV2 does not accept `init_weights` as argument, right ? Let's revert this change.

File: torchvision/models/quantization/shufflenetv2.py [hunk 3]
```diff
@@ -52,6 +52,8 @@ def forward(self, x: Tensor) -> Tensor:
 class QuantizableShuffleNetV2(shufflenetv2.ShuffleNetV2):
     # TODO https://github.com/pytorch/vision/pull/4232#pullrequestreview-730461659
     def __init__(self, *args: Any, **kwargs: Any) -> None:
+        if "init_weights" not in kwargs:
+            kwargs["init_weights"] = True
```
Comment (vfdev-5) [inline torchvision/models/quantization/shufflenetv2.py:56]: Same here, let's revert this change as well

**Comment by puhuk** [comment]
Hmm, why it raises typo error in `test_models.py` when I just revert to origin code

**Comment by vfdev-5** [comment]
@puhuk it is unrelated as main branch is broken : https://app.circleci.com/pipelines/github/pytorch/vision/18344/workflows/d0222980-4837-4c46-a9bf-d9cc4be80051/jobs/1484167

**Comment by puhuk** [comment]
@vfdev-5 Thank you for the feedback!

**Comment by vfdev-5** [comment]
@puhuk there are few similar warnings:
```
test/test_models.py::test_classification_model[cpu-googlenet]
  /root/project/torchvision/models/googlenet.py:47: FutureWarning: The default weight initialization of GoogleNet will be changed in future releases of torchvision. If you wish to keep the old behavior (which leads to long initialization times due to scipy/scipy#11299), please set init_weights=True.
    warnings.warn(

test/test_models.py::test_classification_model[cpu-inception_v3]
  /root/project/torchvision/models/inception.py:43: FutureWarning: The default weight initialization of inception_v3 will be changed in future releases of torchvision. If you wish to keep the old behavior (which leads to long initialization times due to scipy/scipy#11299), please set init_weights=True.
    warnings.warn(
```

Can you try to fix them as well ? Thanks

File: torchvision/models/googlenet.py [hunk 4]
```diff
@@ -34,7 +34,7 @@ def __init__(
         num_classes: int = 1000,
         aux_logits: bool = True,
         transform_input: bool = False,
-        init_weights: Optional[bool] = None,
+        init_weights: bool = True,
```
[the remark this answers is not in the document]
Comment (puhuk) [reply torchvision/models/googlenet.py]: Hmm, I don't have good idea. Is there better method than I did here.
`https://github.com/pytorch/vision/pull/6159/commits/16ddd6c0f23fe56e8029f336c5e3f3c69e9c534f`

File: torchvision/models/quantization/googlenet.py [hunk 5]
```diff
@@ -74,6 +74,8 @@ def forward(self, x: Tensor) -> Tensor:
 class QuantizableGoogLeNet(GoogLeNet):
     # TODO https://github.com/pytorch/vision/pull/4232#pullrequestreview-730461659
     def __init__(self, *args: Any, **kwargs: Any) -> None:
+        if "init_weights" not in kwargs:
+            kwargs["init_weights"] = True
```
Comment (datumbox) [inline torchvision/models/quantization/googlenet.py:78]: I'm not sure why in order to fix the warning on unit-tests we modify the models themselves. My expectations without having all the details in my mind, is that the tests are the ones that need to be updated. What am I missing?

Comment (vfdev-5) [reply torchvision/models/quantization/googlenet.py:78]: At first I was thinking that QuantizableGoogLeNet does not properly set up its base class `GoogLeNet`. Thinking more about that, I think you are right, we have to try again to fix that from tests side

Comment (puhuk) [reply torchvision/models/quantization/googlenet.py:78]: Ok, let me check and send PR again.
